packages/hub/node-tests/@core/babel-plugin-card-schema-transform-test.ts

```diff
@@ -143,6 +143,54 @@ if (process.env.COMPILER) {
       expect(card.compiled.url).to.eq(`${realm}test`);
     });
 
+    it('can handle unconsumed field import for different types of import specifiers', async function () {
+      await cards.create({
+        realm,
+        id: 'bar',
+        schema: 'schema.js',
+        files: {
+          'schema.js': `
+            import { contains } from "@cardstack/types";
+            import string from "https://cardstack.com/base/string";
+
+            export default class Bar {
+              @contains(string) bar;
+            }
+
+            let foo = 'foo';
+            let bar1 = 'bar-1';
+            let bar3 = 'bar-3';
+            export { foo, bar1, bar3 };
+
+            export let bar2 = 'bar-2';
+          `,
+        },
+      });
+
+      let card = await cards.create({
+        realm,
+        id: 'test',
+        schema: 'schema.js',
+        files: {
+          'schema.js': `
+            import { contains } from "@cardstack/types";
+            import string from "https://cardstack.com/base/string";
+            import _, { _camelCase } from 'lodash';
+            import * as randomImport from 'i-dont-exist';
+            import badImport, { incorrectUnusedImport } from 'i-dont-exist';
+            import * as foo from "../bar";
+            import { bar3 } from "../bar";
+            import bar2, { bar1, foo as foo1 } from "../bar";
+
+            export default class Test {
+              @contains(string) name;
+            }
+          `,
+        },
+      });
+      expect(card.compiled.url).to.eq(`${realm}test`);
```

Comment:
we should also assert that in the schema source we have pruned the unconsumed imports

Reply:
Done